### PR TITLE
ESLint and Github on push and on PR action set up

### PR DESCRIPTION
**Pull request by grynevychs**
upgrade ESLint to version 9, fix eslint errors, add Github on push and on PR linting workflow

### CODE REVIEW

File: src/models/index.ts
Comment (learningSherban) [inline src/models/index.ts:1]: Why are you setup relations in index file? I think better to use index file only for reexport

File: .github/workflows/lint.yml
Comment (skelemen-skelar) [inline .github/workflows/lint.yml:1]: great, extra task is done ✅

File: eslint.config.js
Comment (skelemen-skelar) [inline eslint.config.js:1]: Looks like linter was already installed, so almost no lint fixes) You've just refactored linter and now it looks cleaner

File: eslint.config.mjs
Comment (skelemen-skelar) [inline eslint.config.mjs:1]: just curious - why have you chosen mjs extension over js here?

**Review by skelemen-skelar — APPROVED** [review]
PR looks pretty!
Nothing extensive, but:
1. Branch was created with correct naming
2. Linter was refactored and now looks cleaner. Standard presets look justified to me
2.1 Extra task for linting using GHA was done and it works
3. Linter runs without errors
4. Changes were pushed to the branch and PR created

One small note:
maybe some small rule over presets could be added. It would have demonstrated power of linters))
f. e. length of lines 80 is way too small (100 would work I think) but it's more like subjective thing

File: eslint.config.mjs
Comment (grynevychs) [reply eslint.config.mjs:1]: ESLint was complaining about ES modules expecting `require` with eslint.config.js

Screenshot from 2024-06-16 10-11-58

**Pull request merged**